### PR TITLE
Strict Typescript - Dropzone, VisuallyHidden

### PR DESCRIPTION
**Pull request by ktabors**
additional work on #3183 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 🧢 Your Project:
RSP

### CODE REVIEW

File: packages/@react-spectrum/dropzone/stories/DropZone.stories.tsx [hunk 1]
```diff
@@ -29,6 +29,13 @@ import {SpectrumDropZoneProps} from '../src/DropZone';
 import styles from './styles.css';
 import Upload from '@spectrum-icons/illustrations/Upload';
 
+interface FileFilledSource {
+  src?: string;
+  id?: number;
+  type: string;
+  name: string;
+}
+
```
Comment (ktabors) [inline packages/@react-spectrum/dropzone/stories/DropZone.stories.tsx:38]: I'm not happy that I had to declare this interface for a Storybook file.

File: packages/@react-spectrum/dropzone/stories/DropZone.stories.tsx [hunk 2]
```diff
@@ -166,23 +173,23 @@ function Example(props) {
...
     <>
       <Draggable />
-      <DropZone 
+      <DropZone
         {...props}
         isFilled={isFilled}
-        onDrop={async (e) => {
-          let items = await Promise.all(e.items.filter((item) => item.kind === 'text' && item.types.has('text/plain')).map((item: TextDropItem) => item.getText('text/plain')));
+        onDrop={async (e: DropEvent) => {
+          let items = await Promise.all(e.items.filter((item) => item.kind === 'text' && item.types.has('text/plain')).map((item) => (item as TextDropItem).getText('text/plain')));
```
Comment (ktabors) [inline packages/@react-spectrum/dropzone/stories/DropZone.stories.tsx:185]: This came from docs, I think this [example](https://react-spectrum.adobe.com/react-aria/useDrop.html#example). If this is correct we should update those too.

File: tsconfig.json [hunk 3]
```diff
@@ -65,6 +66,8 @@
           "./packages/@react-spectrum/contextualhelp",
           "./packages/@react-spectrum/checkbox",
           "./packages/@react-spectrum/divider",
+          "./packages/@react-spectrum/dropzone",
+          "./packages/@react-spectrum/inlinealert",
```
Comment (ktabors) [inline tsconfig.json:70]: Rob and I have conflicting PRs. I'll probably need to remove this.

File: packages/@react-stately/checkbox/src/useCheckboxGroupState.ts [hunk 4]
```diff
@@ -129,7 +129,7 @@ export function useCheckboxGroupState(props: CheckboxGroupProps = {}): CheckboxG
     },
     validationState: props.validationState ?? (isInvalid ? 'invalid' : null),
     isInvalid,
-    isRequired
+    isRequired: !!isRequired
```
Comment (ktabors) [inline packages/@react-stately/checkbox/src/useCheckboxGroupState.ts:132]: will pull too

File: packages/@react-stately/checkbox/src/useCheckboxGroupState.ts [hunk 5]
```diff
@@ -129,7 +129,7 @@ export function useCheckboxGroupState(props: CheckboxGroupProps = {}): CheckboxG
     },
     validationState: props.validationState ?? (isInvalid ? 'invalid' : null),
     isInvalid,
-    isRequired
+    isRequired: isRequired
```
Comment (snowystinger) [inline packages/@react-stately/checkbox/src/useCheckboxGroupState.ts:132]: why?

File: packages/@react-spectrum/dropzone/stories/DropZone.stories.tsx (hunk 2 above)
Comment (snowystinger) [reply packages/@react-spectrum/dropzone/stories/DropZone.stories.tsx:185]: this is fine, for other examples of ways to solve this, see https://www.skovy.dev/blog/typescript-filter-array-with-type-guard?seed=64zeg1

File: packages/@react-spectrum/dropzone/stories/DropZone.stories.tsx (hunk 1 above)
Comment (snowystinger) [reply packages/@react-spectrum/dropzone/stories/DropZone.stories.tsx:38]: not a problem, also expected and informs us what people will need to do in their own projects in order to use our stuff

File: packages/@react-spectrum/dropzone/stories/DropZone.stories.tsx (hunk 2 above)
Comment (ktabors) [reply packages/@react-spectrum/dropzone/stories/DropZone.stories.tsx:185]: Thanks for sharing. I tried the second suggestion and `let items = await Promise.all(e.items.filter((item): item is TextDropItem => item.kind === 'text' && item.types.has('text/plain')).map((item) => (item).getText('text/plain')));` works and allows us not to set the event type. I couldn't find us using 'is' in any /src/* files in the project. This is very good to know!

**Review by snowystinger — APPROVED** [review]

**Review by yihuiliao — APPROVED** [review]

**Review by yihuiliao — APPROVED** [review]

**Review by snowystinger — APPROVED** [review]

**Pull request merged**